test/hotspot/jtreg/serviceability/dcmd/vm/ClassLoaderStatsTest.java

```diff
@@ -171,12 +171,13 @@ class HiddenClass { }
 class TestClass {
     private static final String HCName = "HiddenClass.class";
     private static final String DIR = System.getProperty("test.classes");
+    public static final Class<?> hc;
 
     static {
         try {
-            // Create a hidden non-strong class
+            // Create a hidden class, keep reference in the case if GC happens
```

Comment:
"Create a hidden class. Keep a reference in case a GC happens."

I hadn't noticed the original comment when first suggesting making the Class reference static. I wonder what was meant by "non-strong class".

Reply:
This line added by  8238358: Implementation of JEP 371: Hidden Classes which has many co-authors. Hope someone could provide an explanation during this review.

It might be possible that the goal was to verify that  VM.classloader_stats provide might provide info for non-reachable clasees. However it makes test to fragile, since can't block class unloading now.

Reply:
Yes, that seems to be the case, and you are right that it is not something that is safe to assume.

Reply:
Lookup.defineHiddenClass allows class options to be specified, one of which is "STRONG" to mean that the hidden class can't unloaded if ts defining loader is reachable.

Reply:
By default hidden classes are non-strong and can be unloaded when no longer referenced, even if their classloader is still reachable. As you want the hidden class to stick around so it can be reported in the stats then either keep a reference (as you have) or else create it as a STRONG hidden class.